libraries/apollo-testing-support/src/commonMain/kotlin/com/apollographql/apollo3/testing/TestNetworkTransport.kt

```diff
@@ -16,40 +20,45 @@ import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.yield
 
-private sealed interface TestResponse {
-  object NetworkError : TestResponse
-  class Response(val response: ApolloResponse<out Operation.Data>) : TestResponse
-}
-
 @ApolloExperimental
 class QueueTestNetworkTransport : NetworkTransport {
   private val lock = reentrantLock()
-  private val queue = ArrayDeque<TestResponse>()
+  private val queue = ArrayDeque<ApolloResponse<out Operation.Data>>()
 
   override fun <D : Operation.Data> execute(request: ApolloRequest<D>): Flow<ApolloResponse<D>> {
     return flow {
       // "Emulate" a network call
       yield()
 
-      val response = lock.withLock { queue.removeFirstOrNull() } ?: error("No more responses in queue")
-      if (response is TestResponse.NetworkError) throw ApolloNetworkException("Network error queued in QueueTestNetworkTransport")
-
       @Suppress("UNCHECKED_CAST")
-      val apolloResponse = (response as TestResponse.Response).response as ApolloResponse<D>
-
+      val apolloResponse = lock.withLock { queue.removeFirstOrNull() } as ApolloResponse<D>? ?: error("No more responses in queue")
```

Comment:
We shouldn't throw in `NetworkTransport.execute`. `"No more responses in queue"` should be in `apolloResponse.exception`

Reply:
I was wondering about this one! My thinking was: if this happens, it's a programming mistake, and so it might be good that it throws? Not a strong opinion though, I'm also 👍 to emit a response with exception.

Reply:
Fair point, Let's go with throwing on programming mistakes 👍